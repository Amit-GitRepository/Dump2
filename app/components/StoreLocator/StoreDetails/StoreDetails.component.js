/* StoreDetails: Shows the details of each store shown in the store locator list view*/

import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import StoreDetailItem from '../StoreDetailItem/StoreDetailItem.component';
import styles from './StoreDetails.style';
import trueShopStoreDetailsTestIDs from '../../../config/testid/trueShopStoreDetails';
import {Button, Card, ISText, List} from '../../Shared';
import {ScrollView, View} from 'react-native';
import {translate} from '../../../language/i18n/helper';

class StoreDetails extends Component {
  formatContact = (contact) => {
    if (contact) {
      // to convert 9 or 10 digit contact number from 0916967185 to 091-696-7185 format with hyphen
      return (contact.substr(0, (contact.length - 7)) + '-' + contact.substr(-7, 3) + '-' + contact.substr(-4));
    }
    return null;
  }

  render () {
    const {onServiceClick, onGetDirectionClick, onCallClick, details} = this.props;
    const {services = [], distance, address, landmark, contact, name} = details;
    return (
      <ScrollView style={styles.container} contentContainerStyle={styles.scrollContainer}>
        <View style={styles.outerContainer} elevation={2}>
          <Card header={name} containerStyles={styles.containerStyles} elevation={0}>
            <View style={styles.detailsContainer}>
              <StoreDetailItem heading={translate('STORE_LOCATOR__DETAILS_DISTANCE')} accessibilityLabel={trueShopStoreDetailsTestIDs.LABEL__DISTANCE} fontWeight={'BOLD'} label={translate('STORE_LOCATOR__DETAILS_DISTANCE_VAL', {distance})}/>
              <StoreDetailItem heading={translate('STORE_LOCATOR__DETAILS_ADDRESS')} accessibilityLabel={trueShopStoreDetailsTestIDs.LABEL__ADDRESS} label={address}/>
              <StoreDetailItem heading={translate('STORE_LOCATOR__DETAILS_LANDMARK')} accessibilityLabel={trueShopStoreDetailsTestIDs.LABEL__LANDMARK} label={landmark}/>
              <StoreDetailItem heading={translate('STORE_LOCATOR__DETAILS_CONTACT')} accessibilityLabel={trueShopStoreDetailsTestIDs.LABEL__CONTACT} label={this.formatContact(contact)}/>
              <View style={styles.buttonsContainer}>
                {contact ? <View style={styles.callButtonWrapper}><Button text={translate('STORE_LOCATOR__DETAILS_CONTACT')} style={styles.buttonCall} textStyle={styles.buttonText} onPress={onCallClick} type={'secondary'} accessibilityLabel={trueShopStoreDetailsTestIDs.BTN__CALL} testID={trueShopStoreDetailsTestIDs.BTN__CALL}/></View> : null}
                <View style={styles.directionButtonWrapper}><Button text={translate('STORE_LOCATOR__DETAILS_GET_DIRECTION')} onPress={onGetDirectionClick} accessibilityLabel={trueShopStoreDetailsTestIDs.BTN__GET_DIRECTION} testID={trueShopStoreDetailsTestIDs.BTN__GET_DIRECTION}/></View>
              </View>
            </View>
            {services.length > 0 ? <ISText style={styles.typeHeading}>{translate('STORE_LOCATOR__DETAILS_TYPE_OF_SERVICE')}</ISText> : null}
            <List data={services} headingKey='serviceName' onPress={onServiceClick}/>
          </Card>
        </View>
      </ScrollView>);
  }
}
StoreDetails.defaultProps = {
  details: {},
  onServiceClick: noop,
  onGetDirectionClick: noop,
  onCallClick: noop
};
StoreDetails.propTypes = {
  details: PropTypes.object,
  onServiceClick: PropTypes.func,
  onGetDirectionClick: PropTypes.func,
  onCallClick: PropTypes.func
};
export default StoreDetails;
