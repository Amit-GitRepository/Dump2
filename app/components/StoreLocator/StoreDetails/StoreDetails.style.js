import {colors, fonts} from '../../../themes/constants.styles';
import {shadowScattered} from '../../../themes/application.styles';

export default {
  container: {
    paddingVertical: 10,
    paddingHorizontal: 5
  },
  scrollContainer: {
    paddingBottom: 20
  },
  detailsContainer: {
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    marginTop: -12
  },
  buttonsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 10
  },
  callButtonWrapper: {
    flex: 1,
    paddingLeft: 16
  },
  directionButtonWrapper: {
    flex: 1,
    paddingHorizontal: 16
  },
  buttonCall: {
    borderColor: colors.PRIMARY_ACTIONABLE
  },
  buttonText: {
    color: colors.PRIMARY_ACTIONABLE
  },
  typeHeading: {
    padding: 5,
    textAlign: 'center',
    backgroundColor: colors.PRIMARY_BG_SEPARATOR,
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  containerStyles: {
    borderRadius: 8,
    borderColor: colors.TRANSPARENT
  },
  outerContainer: {
    borderRadius: 8,
    marginHorizontal: 5,
    ...shadowScattered
  }
};
