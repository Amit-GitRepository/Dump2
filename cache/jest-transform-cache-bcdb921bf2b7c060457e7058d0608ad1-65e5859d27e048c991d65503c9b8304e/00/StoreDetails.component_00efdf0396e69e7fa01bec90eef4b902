Object.defineProperty(exports,"__esModule",{value:true});var cov_qqju5dwze=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreDetails/StoreDetails.component.js',hash='4d7e596ebbc5c44c1e9374b02112d36d9cd3e5b5',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreDetails/StoreDetails.component.js',statementMap:{'0':{start:{line:17,column:4},end:{line:20,column:5}},'1':{start:{line:19,column:6},end:{line:19,column:112}},'2':{start:{line:21,column:4},end:{line:21,column:16}},'3':{start:{line:25,column:72},end:{line:25,column:82}},'4':{start:{line:26,column:72},end:{line:26,column:79}},'5':{start:{line:27,column:4},end:{line:43,column:21}},'6':{start:{line:46,column:0},end:{line:51,column:2}},'7':{start:{line:52,column:0},end:{line:57,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:16,column:18},end:{line:16,column:19}},loc:{start:{line:16,column:31},end:{line:22,column:3}},line:16},'1':{name:'(anonymous_1)',decl:{start:{line:24,column:2},end:{line:24,column:3}},loc:{start:{line:24,column:12},end:{line:44,column:3}},line:24}},branchMap:{'0':{loc:{start:{line:17,column:4},end:{line:20,column:5}},type:'if',locations:[{start:{line:17,column:4},end:{line:20,column:5}},{start:{line:17,column:4},end:{line:20,column:5}}],line:17},'1':{loc:{start:{line:26,column:11},end:{line:26,column:24}},type:'default-arg',locations:[{start:{line:26,column:22},end:{line:26,column:24}}],line:26},'2':{loc:{start:{line:36,column:15},end:{line:36,column:231}},type:'cond-expr',locations:[{start:{line:36,column:25},end:{line:36,column:224}},{start:{line:36,column:227},end:{line:36,column:231}}],line:36},'3':{loc:{start:{line:40,column:11},end:{line:40,column:137}},type:'cond-expr',locations:[{start:{line:40,column:33},end:{line:40,column:130}},{start:{line:40,column:133},end:{line:40,column:137}}],line:40}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0},f:{'0':0,'1':0},b:{'0':[0,0],'1':[0],'2':[0,0],'3':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();

var _Button=require('../../Shared/Button/Button.component');var _Button2=_interopRequireDefault(_Button);
var _Card=require('../../Shared/Card/Card.component');var _Card2=_interopRequireDefault(_Card);
var _ISText=require('../../Shared/ISText/ISText.component');var _ISText2=_interopRequireDefault(_ISText);
var _List=require('../../Shared/List/List.component');var _List2=_interopRequireDefault(_List);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _StoreDetailItem=require('../StoreDetailItem/StoreDetailItem.component');var _StoreDetailItem2=_interopRequireDefault(_StoreDetailItem);
var _StoreDetails=require('./StoreDetails.style');var _StoreDetails2=_interopRequireDefault(_StoreDetails);
var _reactNative=require('react-native');
var _helper=require('../../../language/i18n/helper');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

StoreDetails=function(_Component){_inherits(StoreDetails,_Component);function StoreDetails(){var _ref;var _temp,_this,_ret;_classCallCheck(this,StoreDetails);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=StoreDetails.__proto__||Object.getPrototypeOf(StoreDetails)).call.apply(_ref,[this].concat(args))),_this),_this.
formatContact=function(contact){cov_qqju5dwze.f[0]++;cov_qqju5dwze.s[0]++;
if(contact){cov_qqju5dwze.b[0][0]++;cov_qqju5dwze.s[1]++;

return contact.substr(0,contact.length-7)+'-'+contact.substr(-7,3)+'-'+contact.substr(-4);
}else{cov_qqju5dwze.b[0][1]++;}cov_qqju5dwze.s[2]++;
return null;
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(StoreDetails,[{key:'render',value:function render()

{cov_qqju5dwze.f[1]++;var _ref2=(cov_qqju5dwze.s[3]++,
this.props),onServiceClick=_ref2.onServiceClick,onGetDirectionClick=_ref2.onGetDirectionClick,onCallClick=_ref2.onCallClick,details=_ref2.details;var _ref3=(cov_qqju5dwze.s[4]++,
details),_ref3$services=_ref3.services,services=_ref3$services===undefined?(cov_qqju5dwze.b[1][0]++,[]):_ref3$services,distance=_ref3.distance,address=_ref3.address,landmark=_ref3.landmark,contact=_ref3.contact,name=_ref3.name;cov_qqju5dwze.s[5]++;
return(
_react2.default.createElement(_reactNative.ScrollView,{style:_StoreDetails2.default.container,contentContainerStyle:_StoreDetails2.default.scrollContainer},
_react2.default.createElement(_Card2.default,{header:name},
_react2.default.createElement(_reactNative.View,{style:_StoreDetails2.default.detailsContainer},
_react2.default.createElement(_StoreDetailItem2.default,{heading:(0,_helper.translate)('STORE_LOCATOR__DETAILS_DISTANCE'),fontWeight:'BOLD',label:(0,_helper.translate)('STORE_LOCATOR__DETAILS_DISTANCE_VAL',{distance:distance})}),
_react2.default.createElement(_StoreDetailItem2.default,{heading:(0,_helper.translate)('STORE_LOCATOR__DETAILS_ADDRESS'),label:address}),
_react2.default.createElement(_StoreDetailItem2.default,{heading:(0,_helper.translate)('STORE_LOCATOR__DETAILS_LANDMARK'),label:landmark}),
_react2.default.createElement(_StoreDetailItem2.default,{heading:(0,_helper.translate)('STORE_LOCATOR__DETAILS_CONTACT'),label:this.formatContact(contact)}),
_react2.default.createElement(_reactNative.View,{style:_StoreDetails2.default.buttonsContainer},
contact?(cov_qqju5dwze.b[2][0]++,_react2.default.createElement(_reactNative.View,{style:_StoreDetails2.default.buttonWrapper},_react2.default.createElement(_Button2.default,{text:(0,_helper.translate)('STORE_LOCATOR__DETAILS_CONTACT'),style:_StoreDetails2.default.buttonCall,textStyle:_StoreDetails2.default.buttonText,onPress:onCallClick,type:'secondary'}))):(cov_qqju5dwze.b[2][1]++,null),
_react2.default.createElement(_reactNative.View,{style:_StoreDetails2.default.buttonWrapper},_react2.default.createElement(_Button2.default,{text:(0,_helper.translate)('STORE_LOCATOR__DETAILS_GET_DIRECTION'),onPress:onGetDirectionClick})))),


services.length>0?(cov_qqju5dwze.b[3][0]++,_react2.default.createElement(_ISText2.default,{style:_StoreDetails2.default.typeHeading},(0,_helper.translate)('STORE_LOCATOR__DETAILS_TYPE_OF_SERVICE'))):(cov_qqju5dwze.b[3][1]++,null),
_react2.default.createElement(_List2.default,{data:services,headingKey:'serviceName',onPress:onServiceClick}))));


}}]);return StoreDetails;}(_react.Component);cov_qqju5dwze.s[6]++;

StoreDetails.defaultProps={
details:{},
onServiceClick:_noop2.default,
onGetDirectionClick:_noop2.default,
onCallClick:_noop2.default};cov_qqju5dwze.s[7]++;

StoreDetails.propTypes={
details:_propTypes2.default.object,
onServiceClick:_propTypes2.default.func,
onGetDirectionClick:_propTypes2.default.func,
onCallClick:_propTypes2.default.func};exports.default=

StoreDetails;