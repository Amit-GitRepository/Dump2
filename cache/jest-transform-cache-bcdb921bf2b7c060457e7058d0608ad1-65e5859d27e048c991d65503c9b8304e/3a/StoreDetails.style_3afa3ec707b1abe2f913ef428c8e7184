Object.defineProperty(exports,"__esModule",{value:true});var cov_1wqqcdlunt=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreDetails/StoreDetails.style.js',hash='dafcc8eb16fbe60129cddc2a5238059ca7f759f6',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreDetails/StoreDetails.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
container:{
padding:10},

scrollContainer:{
paddingBottom:20},

detailsContainer:{
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},

buttonsContainer:{
flexDirection:'row',
justifyContent:'space-between',
paddingVertical:10},

buttonWrapper:{
flex:1,
paddingHorizontal:20},

buttonCall:{
borderColor:_constants.colors.PRIMARY_ACTIONABLE},

buttonText:{
color:_constants.colors.PRIMARY_ACTIONABLE},

typeHeading:{
padding:5,
textAlign:'center',
backgroundColor:_constants.colors.PRIMARY_BG_SEPARATOR,
fontSize:_constants.fonts.FONT_SIZE_MEDIUM},

containerStyles:{
borderRadius:5,
borderBottomWidth:5,
borderColor:_constants.colors.TRANSPARENT}};