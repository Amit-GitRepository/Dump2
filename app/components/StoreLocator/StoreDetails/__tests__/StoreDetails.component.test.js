import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import StoreDetails from '../StoreDetails.component';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('StoreDetails component', () => {
  it('StoreDetails: renders correctly', () => {
    const tree = renderer.create(<StoreDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('formatContact: returns the contact number in correct format for 9 digit contact number', () => {
    const wrapper = shallow(<StoreDetails />);
    expect(wrapper.instance().formatContact('026584088')).toEqual('02-658-4088');
  });

  it('formatContact: returns the contact number in correct format for 10 digit contact number', () => {
    const wrapper = shallow(<StoreDetails />);
    expect(wrapper.instance().formatContact('0916967185')).toEqual('091-696-7185');
  });

  it('formatContact: returns null if null is passed', () => {
    const wrapper = shallow(<StoreDetails />);
    expect(wrapper.instance().formatContact(null)).toEqual(null);
  });

});
