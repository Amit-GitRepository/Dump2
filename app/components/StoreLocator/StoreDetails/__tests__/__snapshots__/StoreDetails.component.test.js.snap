// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreDetails component StoreDetails: renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 20,
    }
  }
  style={
    Object {
      "paddingHorizontal": 5,
      "paddingVertical": 10,
    }
  }
>
  <View>
    <View
      elevation={2}
      style={
        Object {
          "borderRadius": 8,
          "marginHorizontal": 5,
          "shadowColor": "#303C46",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 2,
        }
      }
    >
      <View
        elevation={0}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flex": 1,
              "marginHorizontal": -1,
              "overflow": "hidden",
              "shadowColor": "#303C46",
              "shadowOffset": Object {
                "height": 2,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            },
            Object {
              "borderColor": "transparent",
              "borderRadius": 8,
            },
          ]
        }
      >
        <View
          onPress={null}
          style={
            Array [
              Object {
                "zIndex": 1,
              },
              Object {},
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "borderWidth": 0,
                  "elevation": 3,
                  "shadowColor": "#303C46",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomColor": "#F93E60",
                      "paddingHorizontal": 5,
                    },
                    Object {
                      "borderBottomColor": "#303C46",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#E9E9EA",
                          "padding": 10,
                        },
                        Object {
                          "backgroundColor": "#303C46",
                          "marginHorizontal": -5,
                          "padding": 10,
                        },
                      ],
                      Object {},
                    ]
                  }
                />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#303C46",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "bottom": 9.6,
                      "marginBottom": -9.6,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              Ó§Å
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "marginTop": -12,
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#E9E9EA",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "padding": 15,
              }
            }
          >
            <Text
              accessibilityLabel="trueShopStoreDetails_label_distance_heading"
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "flex": 1,
                  "fontFamily": "SukhumvitSet-Bold",
                  "fontSize": 18,
                }
              }
              testID="trueShopStoreDetails_label_distance_heading"
            >
              Distance
            </Text>
            <Text
              accessibilityLabel="trueShopStoreDetails_label_distance_value"
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "flex": 1,
                  "fontFamily": "SukhumvitSet-Bold",
                  "fontSize": 18,
                }
              }
              testID="trueShopStoreDetails_label_distance_value"
            >
              [missing {{distance}} value] k.m.
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingVertical": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel="trueShopStoreDetails_button_get_direction"
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#F93E60",
                    "borderRadius": 50,
                    "opacity": 1,
                    "padding": 10,
                  }
                }
                testID="trueShopStoreDetails_button_get_direction"
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "stretch",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#303C46",
                          },
                          Array [
                            Object {
                              "color": "#FFFFFF",
                              "fontSize": 16,
                            },
                            Object {},
                          ],
                          Object {
                            "fontFamily": "SukhumvitSet-Bold",
                          },
                        ]
                      }
                    >
                      Get Direction
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
